src/overlays/actors/ovl_En_Bb/z_en_bb.c

```diff
@@ -32,65 +105,1137 @@ const ActorInit En_Bb_InitVars = {
+    Math_SmoothStepToF(&this->flameScaleX, 100.0f, 1.0f, 10.0f, 0.0f);
+    if (this->actor.groundY > -32000.0f) {
+        Math_SmoothStepToF(&this->actor.posRot.pos.y, this->actor.groundY + 50.0f + this->flyHeightMod, 1.0f, 0.5f,
+                                0.0f);
+    }
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (Math_CosF(this->bobPhase) == 0.0f) {
+        if (this->charge) {
+            this->bobSpeedMod = Rand_ZeroOne() * 2.0f;
+        } else {
+            this->bobSpeedMod = Rand_ZeroOne() * 4.0f;
+        }
+    }
+    this->actor.posRot.pos.y += (Math_CosF(this->bobPhase) * (1.0f + this->bobSpeedMod));
+    this->bobPhase += 0.2f;
+    Math_SmoothStepToF(&this->actor.speedXZ, this->maxSpeed, 1.0f, 0.5f, 0.0f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bb/func_809BA628.s")
+    if (Math_Vec3f_DistXZ(&this->actor.posRot.pos, &this->actor.initPosRot.pos) > 300.0f) {
+        this->moveAngleY = Math_Vec3f_Yaw(&this->actor.posRot.pos, &this->actor.initPosRot.pos);
+        Math_SmoothStepToS(&this->actor.posRot.rot.y, this->moveAngleY, 1, 0x7D0, 0);
+    } else {
+        this->timer--;
+        if (this->timer <= 0) {
+            this->charge ^= true;
+            this->flyHeightMod = (s16)(Math_CosF(this->bobPhase) * 10.0f);
+            this->actor.speedXZ = 0.0f;
+            if (this->charge && (this->targetActor == NULL)) {
+                this->moveAngleY = this->actor.posRot.rot.y;
+                if (this->actor.xzDistFromLink < 200.0f) {
+                    SkelAnime_ChangeAnimDefaultRepeat(&this->skelAnime, &D_06000184);
+                    this->moveAngleY = this->actor.yawTowardsLink;
+                }
+                this->maxSpeed = Rand_ZeroOne() * 1.5f + 6.0f;
+                this->timer = Rand_ZeroOne() * 5.0f + 20.0f;
+                this->actionState = false;
+            } else {
+                SkelAnime_ChangeAnimDefaultRepeat(&this->skelAnime, &D_06000444);
+                this->maxSpeed = (Rand_ZeroOne() * 1.5f) + 1.0f;
+                this->timer = (Rand_ZeroOne() * 20.0f) + 40.0f;
+                this->moveAngleY = Math_SinF(this->bobPhase) * 65535.0f;
+            }
+        }
+        if ((this->actor.xzDistFromLink < 150.0f) && this->actionState) {
+            if (!this->charge) {
+                SkelAnime_ChangeAnimDefaultRepeat(&this->skelAnime, &D_06000184);
+                this->maxSpeed = Rand_ZeroOne() * 1.5f + 6.0f;
+                this->timer = Rand_ZeroOne() * 5.0f + 20.0f;
+                this->moveAngleY = this->actor.yawTowardsLink;
+                this->actionState = this->charge = true;
+            }
+        } else if (this->actor.xzDistFromLink < 200.0f) {
+            this->moveAngleY = this->actor.yawTowardsLink;
+        }
+        if (this->targetActor == NULL) {
+            bomb = EnBb_FindExplosive(globalCtx, this, 300.0f);
+        } else if (this->targetActor->params == 0) {
```

Comment:
this can be a chu as well right? would change the temp names here and in the find function to be for explosive explicitly